lib/compiler.js

```diff
@@ -186,39 +186,82 @@ Compiler.prototype.visitTag = function (tag) {
  * Also, `virtual-dom` handles properties differently.
  */
 
+// attribute -> property transformations
+const ATTRS_TO_PROPS = {
+  'class': 'className',
+  'for': 'htmlFor',
+  'http-equiv': 'httpEquiv',
+}
+const LOWERCASE_ATTRS = new Set([
+  'acceptCharset',
+  'accessKey',
+  'allowFullScreen',
+  'allowTransparency',
+  'cellPadding',
+  'cellSpacing',
+  'colSpan',
+  'contentEditable',
+  'contextMenu',
+  'crossOrigin',
+  'dateTime',
+  'formAction',
+  'formEncType',
+  'formMethod',
+  'formNoValidate',
+  'formTarget',
+  'frameBorder',
+  'marginHeight',
+  'marginWidth',
+  'maxLength',
+  'mediaGroup',
+  'noValidate',
+  'readOnly',
+  'rowSpan',
+  'tabIndex',
+  'useMap',
+])
+for (const a of LOWERCASE_ATTRS) {
+  ATTRS_TO_PROPS[a.toLowerCase()] = a
+}
+
+function dashedToCaps(s) {
+  return s[1].toUpperCase()
```

Comment:
With the global flag, would the second argument not always be what you want?

On Monday, June 20, 2016, tdumitrescu <notifications@github.com> wrote:

> In lib/compiler.js
> <https://github.com/tdumitrescu/virtual-jade/pull/12#discussion_r67716394>
> :
>
> > +  'marginHeight',
> > +  'marginWidth',
> > +  'maxLength',
> > +  'mediaGroup',
> > +  'noValidate',
> > +  'readOnly',
> > +  'rowSpan',
> > +  'tabIndex',
> > +  'useMap',
> > +])
> > +for (const a of LOWERCASE_ATTRS) {
> > +  ATTRS_TO_PROPS[a.toLowerCase()] = a
> > +}
> > +
> > +function dashedToCaps(s) {
> > +  return s[1].toUpperCase()
>
> the problem with that is that i wouldn't be able to pass the function
> directly to .replace(), which then defeats the purpose of extracting it
> in the first place (getting the linter to shut up about creating functions
> inside of loops). But you're right, it's unreadable garbage and I don't
> actually care about compile speed micro-optimizations, so I just extracted
> the whole dashedToCamel function in 0a1e17d
> <https://github.com/tdumitrescu/virtual-jade/pull/12/commits/0a1e17d4671f953073ee54cbb7f35e5d950c399f>
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tdumitrescu/virtual-jade/pull/12/files/18134f6a491f838d4ef3e484894725dd63b73fd1#r67716394>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAh_Ut10MDHIAZn5ou506Yzn-C07rLJ_ks5qNrnCgaJpZM4I5V2Y>
> .
>